float64 lw_velocity
float64 rw_velocity